kolibri/plugins/coach/assets/src/state/actions/reports.js

```diff
@@ -131,107 +173,179 @@ function _userSummaryState(data) {
+    collection_id: options.classId,
+  };
+  const promises = [
+    _setContentSummary(store, options.contentScopeId, reportPayload),
+    _setLearnerReport(store, reportPayload),
+  ];
+  Promise.all(promises).then(
+    () => {
+      const reportProps = {
+        classId: options.classId,
+        channelId: options.channelId,
+        contentScope: options.contentScope,
+        contentScopeId: options.contentScopeId,
+        userScope: options.userScope,
+        userScopeId: options.userScopeId,
+        viewBy: ReportConstants.ViewBy.LEARNER,
+      };
+      store.dispatch('SET_REPORT_PROPERTIES', reportProps);
+      store.dispatch('CORE_SET_PAGE_LOADING', false);
     },
     error => coreActions.handleError(store, error)
   );
 }
 
+// needs exercise, attemptlog. Pass answerstate into contentrender to display answer
+function _showExerciseDetailView(store, classId, userId, channelId, contentId,
+  questionNumber, interactionIndex) {
+  console.log('arrived in our action');
+  Promise.all([
+    ContentNodeResource.getCollection({ channel_id: channelId }, { content_id: contentId }).fetch(),
+    AttemptLogResource.getCollection({ user: userId, content: contentId }).fetch(),
+    SummaryLogResource.getCollection({ user_id: userId, content_id: contentId }).fetch(),
+    FacilityUserResource.getModel(userId).fetch(),
+  ]).then(
+    ([exercises, attemptLogs, summaryLog, user]) => {
+      function parseJSONorUndefined(json) {
+        try {
+          return JSON.parse(json);
+        } catch (e) {
+          if (!(e instanceof SyntaxError)) {
+            throw e;
+          }
+        }
+        return undefined;
+      }
+      // MAPPERS NEEDED
+      // attemptLogState
+      // attemptLogListState
+      // interactionState
+      // InteractionHistoryState
+      // user?
+
+      const exercise = exercises[0];
+
+      // FIRST LOOP: Sort them by most recent
+      attemptLogs.sort(
+        (attemptLog1, attemptLog2) =>
+          new Date(attemptLog2.end_timestamp) - new Date(attemptLog1.end_timestamp)
+      );
 
-function _showTopic(store, classId, channelId, topicId) {
-  _showReport(store, {
-    classId,
-    channelId,
-    contentScope: ReportConstants.ContentScopes.TOPIC,
-    contentScopeId: topicId,
-    userScope: ReportConstants.UserScopes.CLASSROOM,
-    userScopeId: classId,
-    sortColumn: ReportConstants.TableColumns.NAME,
-    sortOrder: ReportConstants.SortOrders.NONE,
-  });
+      const exerciseQuestions = parseJSONorUndefined(
+        exercise.assessmentmetadata[0].assessment_item_ids
+      );
+      // SECOND LOOP: Add their question number
+      if (exerciseQuestions) {
+        attemptLogs.forEach(
+          attemptLog => {
+            attemptLog.questionNumber = (exerciseQuestions.indexOf(attemptLog.item) + 1);
+          }
+        );
+      }
+
+
+      // THIRD LOOP: Return the current attempt
+      const currentAttemptLog = () => {
+        if (questionNumber) {
```

Comment:
what condition is intended to be checked here?

Question number might be any number including `0`, and presumably `0` is a valid index? It could also be `undefined` (or `NaN` for that matter because of `Number(undefined)` in app.js)